File: app/callkit-java/src/main/java/com/example/callkit_java/HandleCallReceiver.java
package com.example.callkit_java;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class HandleCallReceiver extends BroadcastReceiver {

    public static final String ACTION_CREATE_CALL = "com.example.callkit.ACTION_CREATE_CALL";
    public static final String ACTION_ACCEPT_CALL = "com.example.callkit.ACTION_ACCEPT_CALL";
    public static final String ACTION_CANCEL_CALL = "com.example.callkit.ACTION_CANCEL_CALL";


    @Override
    public void onReceive(Context context, Intent intent) {
        switch (intent.getAction()) {
            case ACTION_ACCEPT_CALL: {
                CallKit.getInstance(context).setCallStatus(1);
                CallKit.getInstance(context).closeIncomingCallScreen();
                break;
            }
            case ACTION_CANCEL_CALL: {
                CallKit.getInstance(context).setCallStatus(0);
                CallKit.getInstance(context).closeIncomingCallScreen();

            }
            default:
                break;
        }
    }
}

File: app/callkit/src/main/java/com/example/callkit/HandleCallReceiver.kt
package com.example.callkit;

import android.content.BroadcastReceiver;
import android.content.Context
import android.content.Intent

class HandleCallReceiver : BroadcastReceiver() {
    companion object {
        const val ACTION_CREATE_CALL = "com.example.callkit.ACTION_CREATE_CALL";
        const val ACTION_ACCEPT_CALL = "com.example.callkit.ACTION_ACCEPT_CALL";
        const val ACTION_CANCEL_CALL = "com.example.callkit.ACTION_CANCEL_CALL";
    }

    override fun onReceive(context: Context, intent: Intent) {
        when (intent.action ?: "") {
            ACTION_ACCEPT_CALL -> {
                CallKit.getInstance(context).setCallStatus(1)
                CallKit.getInstance(context).closeIncomingCallScreen()
            }
            ACTION_CANCEL_CALL -> {
                CallKit.getInstance(context).setCallStatus(0)
                CallKit.getInstance(context).closeIncomingCallScreen()
            }
        }
    }
}
